src/sdl2_compat.c

```diff
@@ -684,18 +720,100 @@ SDL_GetHintBoolean(const char *name, SDL2_bool default_value)
     return SDL3_GetHintBoolean(SDL2_to_SDL3_hint(name), default_value) ? SDL2_TRUE : SDL2_FALSE;
 }
 
-/* FIXME: callbacks may need tweaking ... */
+static void SDLCALL
+hint_callback_wrapper(void *userdata, const char *name, const char *oldValue, const char *newValue)
+{
+    HintCallbackInfo *callback_info = (HintCallbackInfo *) userdata;
+    const char *sdl2OldValue, *sdl2NewValue;
+    bool freeSdl2OldValue, freeSdl2NewValue;
+
+    sdl2OldValue = SDL3_to_SDL2_hint_value(name, oldValue, &freeSdl2OldValue);
+    sdl2NewValue = SDL3_to_SDL2_hint_value(name, newValue, &freeSdl2NewValue);
+    callback_info->callback(callback_info->userdata, callback_info->name, sdl2OldValue, sdl2NewValue);
+    if (freeSdl2OldValue) {
+        SDL3_free((void *)sdl2OldValue);
+    }
+    if (freeSdl2NewValue) {
+        SDL3_free((void *)sdl2NewValue);
+    }
+}
+
+static char *
+CreateHintCallbackPropertyName(const char *name, SDL_HintCallback callback, void *userdata) {
+    size_t len_property_name = SDL_strlen(name) + 1 + 2 * sizeof(callback) + 1 + 2 * sizeof(userdata) + 1;
+    char *property_name = (char *)SDL3_malloc(len_property_name);
+    if (!property_name) {
+        return NULL;
+    }
+    SDL3_snprintf(property_name, len_property_name, "%s_%x_%x", name, (void *)callback, (void *)userdata);
```

Comment:
We want `%p` instead of `%x` because the width will depend on the bitness of the platform.

Reply:
I did use `%x` because we know an upper bound on the required length (`2 * sizeof(void *)`).
When using `%p`, it depends on the platform whether you get a `0x` prefix or none.

Reply:
You're probably right: `%x`will treat the argument as a 32-bit integer where we need more bits.